scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -48,19 +51,42 @@
     (swap! map-atom dissoc key)
     (swap! map-atom assoc key value)))
 
-;; REVIEW: Review handle-status-update
+(defn container-status->failure-reason
+  [^V1PodStatus pod-status ^V1ContainerStatus status]
+  ; TODO map kubernetes failure reasons
+  (let [terminated (-> status .getState .getTerminated)]
+    (cond
+      (= "OutOfMemory" (.getReason pod-status)) :reason-container-limitation-memory
+      (= "Error" (.getReason terminated)) :reason-command-executor-failed
+      (= "OOMKilled" (.getReason terminated)) :reason-container-limitation-memory
+      :default :unknown)))
```

Comment:
Log here, then we can grep the log for weird states and add them into the case machine.

Reply:
Done